randutil/rand.go

```diff
@@ -32,34 +34,72 @@ import (
 	"time"
 )
 
-func init() {
-	// golang does not init Seed() itself
-	rand.Seed(time.Now().UnixNano() + int64(os.Getpid()))
+var defaultSeeder = SeedDatePid
+
+// SeedDatePid provides a basic seed value based only on
+// time and os PID value.
+func SeedDatePid() int64 {
+	return time.Now().UnixNano() + int64(os.Getpid())
 }
 
-var moreMixedSeedOnce sync.Once
-
-func moreMixedSeed() {
-	moreMixedSeedOnce.Do(func() {
-		h := sha256.New224()
-		// do this instead of asking for time and pid again
-		var b [8]byte
-		rand.Read(b[:])
-		h.Write(b[:])
-		// mix in the hostname
-		if hostname, err := os.Hostname(); err == nil {
-			h.Write([]byte(hostname))
-		}
-		// mix in net interfaces hw addresses (MACs etc)
-		if ifaces, err := net.Interfaces(); err == nil {
-			for _, iface := range ifaces {
-				h.Write(iface.HardwareAddr)
-			}
+// SeedDatePidHostMac provides a seed value that in addition to
+// time and os PID, also takes into account the device hostname
+// and network inteface MAC addresses. This may be required when
+// you are trying to randomize the time of actions within a
+// fleet of similar devices, where the time and PID values may
+// not provide enough variance within the device pool.
+func SeedDatePidHostMac() int64 {
+	// Use a pseudo RNG initially for time and pid inclusion
+	var b [8]byte
+	pr := NewPseudoRand(SeedDatePid)
+	pr.rand.Read(b[:])
+
+	h := sha256.New224()
+	// Mix in the time and pid
+	h.Write(b[:])
+	// Mix in the hostname
+	if hostname, err := os.Hostname(); err == nil {
+		h.Write([]byte(hostname))
+	}
+	// Mix in net interfaces hw addresses (MACs etc)
+	if ifaces, err := net.Interfaces(); err == nil {
+		for _, iface := range ifaces {
+			h.Write(iface.HardwareAddr)
 		}
-		hs := h.Sum(nil)
-		s := binary.LittleEndian.Uint64(hs[0:])
-		rand.Seed(int64(s))
-	})
+	}
+	hs := h.Sum(nil)
+	s := binary.LittleEndian.Uint64(hs[0:])
+	return int64(s)
+}
+
+// PseudoRand provides a go-routine safe randomisation helper methods.
```

Comment:
Nits: 1) per the Go spec, "goroutine" is a single unhyphenated word, 2) the word "a" should be dropped.